### PR TITLE
Survey submit all

### PR DESCRIPTION
**Pull request by susanm74**

### CODE REVIEW

File: grails-app/services/org/chai/kevin/survey/SurveyPageService.java [hunk 1]
```diff
@@ -717,12 +703,51 @@ public void setDataService(DataService dataService) {
...
+	
+	public void setReportService(ReportService reportService) {
+		this.reportService = reportService;
+	}
+	
 	public void setGrailsApplication(GrailsApplication grailsApplication) {
 		this.grailsApplication = grailsApplication;
 	}
 	
+	public Set<String> getSubmitSkipLevels() {
```
Comment (fterrier) [inline grails-app/services/org/chai/kevin/survey/SurveyPageService.java]: does this need to be public ? is it needed at all ?

File: grails-app/services/org/chai/kevin/survey/SurveyPageService.java [hunk 2]
```diff
@@ -717,12 +703,51 @@ public void setDataService(DataService dataService) {
...
+		return submitSkipLevels;
+	}
+
+	public void setSubmitSkipLevels(Set<String> submitSkipLevels) {
+		this.submitSkipLevels = submitSkipLevels;
+	}
+	
+	public Set<LocationLevel> getSkipSubmitLevels() {
```
Comment (fterrier) [inline grails-app/services/org/chai/kevin/survey/SurveyPageService.java:722]: is this really public ?

File: grails-app/services/org/chai/kevin/survey/SurveyPageService.java [hunk 3]
```diff
@@ -717,12 +703,51 @@ public void setDataService(DataService dataService) {
...
+		Set<LocationLevel> levels = new HashSet<LocationLevel>();
+		for (String skipLevel : this.submitSkipLevels) {
+			levels.add(locationService.findLocationLevelByCode(skipLevel));
+		}
+		return levels;
+	}
+	
+	public void getLocationSkipLevels(Set<String> skipLevels) {
```
Comment (fterrier) [inline grails-app/services/org/chai/kevin/survey/SurveyPageService.java]: does this need to be public ?

File: grails-app/services/org/chai/kevin/survey/SurveyPageService.java [hunk 4]
```diff
@@ -717,12 +703,51 @@ public void setDataService(DataService dataService) {
...
+		this.locationSkipLevels = skipLevels;
+	}
+	
+	public void setLocationSkipLevels(Set<String> skipLevels) {
+		this.locationSkipLevels = skipLevels;
+	}
+	
+	public Set<LocationLevel> getSkipLocationLevels() {
```
Comment (fterrier) [inline grails-app/services/org/chai/kevin/survey/SurveyPageService.java:734]: does this need to be public ?

**Comment by fterrier** [comment]
probably those few getters should be private or are not even needed at all, otherwise everything looks GO

**Comment by susanm74** [comment]
Yep, all great catches, thx! I'll make those changes to remove unused getters, and to make the others
private, as they should be!

On Wed, May 23, 2012 at 9:21 AM, François Terrier <
reply@reply.github.com

**Pull request merged**